File: GFG/Python/2025-08-16_form_largest_number.py
class Solution:
    def findLargest(self, arr):
        # Convert all integers to strings
        arr_str = list(map(str, arr))
        
        # Sort using custom rule: compare concatenations
        arr_str.sort(key=lambda x: x*10, reverse=True)  # multiply to avoid length mismatch

        # If the largest number is "0", the whole number is "0"
        if arr_str[0] == "0":
            return "0"
        
        # Concatenate and return
        return ''.join(arr_str)

File: GFG/C++/20250816_form_largest_number.cpp
class Solution {
public:
    string findLargest(vector<int> &arr) {
        // Custom comparator
        auto cmp = [](int &a, int &b) {
            string s1 = to_string(a);
            string s2 = to_string(b);
            return s1 + s2 > s2 + s1; // Compare concatenated results
        };
        
        // Sort with custom comparator
        sort(arr.begin(), arr.end(), cmp);

        // If the first element is 0, result is "0"
        if (arr[0] == 0) return "0";
        
        string res = "";
        for (int num : arr) {
            res += to_string(num);
        }
        return res;
    }
};
